Sources/TootSDK/TootClient/TootClient+TimeLine.swift

```diff
@@ -55,12 +56,39 @@ public extension TootClient {
     /// - Parameters:
     ///   - pageInfo: a PageInfo struct that tells the API how to page the response, typically with a minId set of the highest id you last saw
     ///   - limit: Maximum number of results to return (defaults to 20 on Mastodon with a max of 40)
+    ///   - onlyMedia: Return only statuses with media attachments
     /// - Returns: a PagedResult containing the posts retrieved
-    func getFederatedTimeline(_ pageInfo: PagedInfo? = nil, limit: Int? = nil) async throws -> PagedResult<[Post]> {
+    func getFederatedTimeline(_ pageInfo: PagedInfo? = nil, limit: Int? = nil, onlyMedia: Bool? = nil) async throws -> PagedResult<[Post]> {
         let req = HTTPRequestBuilder {
             $0.url = getURL(["api", "v1", "timelines", "public"])
             $0.method = .get
-            $0.query = getQueryParams(pageInfo, limit: limit)
+            $0.query = getQueryParams(pageInfo, limit: limit, onlyMedia: onlyMedia)
+        }
+        return try await getPosts(req, pageInfo, limit)
+    }
+    
+    /// Retrieves public statuses containing the given hashtag
+    /// - Parameters:
+    ///   - query: a ``HashtagTimelineQuery`` struct defining the parameters of the request
+    ///   - pageInfo: a PageInfo struct that tells the API how to page the response, typically with a minId set of the highest id you last saw
+    ///   - limit: Maximum number of results to return (defaults to 20 on Mastodon with a max of 40)
+    /// - Returns: a PagedResult containing the posts retrieved
+    func getHashtagTimeline(query: HashtagTimelineQuery, _ pageInfo: PagedInfo? = nil, limit: Int? = nil) async throws -> PagedResult<[Post]> {
```

Comment:
To keep this consistent with other timeline requests, let's formulate it as:
```suggestion
    func getHashtagTimeline(_ query: HashtagTimelineQuery, _ pageInfo: PagedInfo? = nil, limit: Int? = nil) async throws -> PagedResult<[Post]> {
```